### PR TITLE
add docker.compose.yml

### PR DESCRIPTION
**Pull request by pavlenex**
Running the website in multiple environments can be troublesome. Hopefully, Docker makes it more env-agnostic.

This PR is based on @johnsBeharry work, I've just put it together and tailored it to my environment.

## To test:

- Install docker https://www.docker.com/products/docker-desktop
- cd into a directory
- Run `docker-compose up` (you may need to wait for 5-10 minutes

This is based on #185 and makes #185 obsolete but #185 needs some testing on its own, so please do not merge this unless people tested #185 and verified there're no troubles with it.

Screenshot 2021-03-19 at 12 22 17

## To-do

If tested & accepted by maintainers and the community, it would be nice to add instructions to the readme.

### CODE REVIEW

**Comment by pavlenex** [comment]
Now that #185 is merged, if we still want to have Docker which is good for cross-environments  this PR would probably need someone to test it out.

**Comment by pavlenex** [comment]
Did one more test and it worked. 

First time you run it if you see `Attaching to guide_jekyll_1` hanging in terminal, just give it some time ~5min-10min to be sure.

**Comment by pavlenex** [comment]
I've added instructions in the `readme.md` on how to run with Docker. I think this one is ready to go.

**Comment by GBKS** [comment]
Just tested it and it worked great. Only thing that tripped me up momentarily was that I had to run `docker compose up`. Above,  and in the documentation, it reads `docker-compose up` (`-` instead of space). Both commands seem to do something, but not sure what the difference is.

**Comment by johnsBeharry** [comment]
i believe it should in-fact be `docker-compose up` unless there is a new sub command that i'm not aware of.

**Review by GBKS — APPROVED** [review]

File: Gemfile.lock [hunk 1]
```diff
@@ -43,6 +43,9 @@ GEM
     image_optim_pack (0.7.0)
       fspath (>= 2.1, < 4)
       image_optim (~> 0.19)
+    image_optim_pack (0.7.0-x86_64-linux)
```
[the remark this answers is not in the document]
Comment (GBKS) [reply Gemfile.lock:46]: @johnsBeharry how should I understand your comment. Is it something we need to fix?

**Comment by pavlenex** [comment]
@johnsBeharry Waiting for your comment about this https://github.com/BitcoinDesign/Guide/pull/186#discussion_r603469188 before merging it.

File: Gemfile.lock (hunk 1 above)
[the remark this answers is not in the document]
Comment (johnsBeharry) [reply Gemfile.lock:46]: I was just pointing it out, didn't notice before that it updated the Gemfile.lock. Can you check if your Gemfile.lock was modified after you run `docker-compose up`? You may have to delete the container so it does the install from scratch.

**Comment by johnsBeharry** [comment]
Ok, so tested again - and anyone running the site in the docker container should just be mindful not to commit the Gemfile.lock if it gets modified by the bundle install that happens in the container.

**Comment by pavlenex** [comment]
@johnsBeharry So what do you suggest I should do, it's not quite clear to me, remove that or keep it as is?

**Review by johnsBeharry — APPROVED** [review]
Ok looks good with this update.

**Pull request merged**